src/Widgets/LinkListBox.vala

```diff
@@ -0,0 +1,286 @@
+                    }
+                    break;
+                case "proprietary":
+                    license_copy = _("Proprietary");
+                    break;
+                default:
+                    license_copy = _("Unknown License");
+                    break;
+            }
+        } else {
+            license_copy = AppStream.get_license_name (project_license);
+            license_url = AppStream.get_license_url (project_license);
+        }
+    }
+
+    private class LinkRow : Gtk.ListBoxRow {
+        public string uri_or_key { get; construct; }
+        public string icon_name { get; construct; }
+        public string label_string { get; construct; }
+        public string color { get; construct; }
+
+        public LinkRow (string uri_or_key, string label_string, string icon_name, string color) {
+            Object (
+                uri_or_key: uri_or_key,
+                label_string: label_string,
+                icon_name: icon_name,
+                color: color
+            );
+        }
+
+        class construct {
+            set_accessible_role (LINK);
+        }
+
+        construct {
+            var image = new Gtk.Image.from_icon_name (icon_name);
+            image.add_css_class (Granite.STYLE_CLASS_ACCENT);
+            image.add_css_class (color);
+
+            var left_label = new Gtk.Label (label_string) {
+                hexpand = true,
+                xalign = 0
+            };
+
+            var link_image = new Gtk.Image.from_icon_name ("adw-external-link-symbolic");
+
+            var box = new Gtk.Box (HORIZONTAL, 0);
+            box.append (image);
+            box.append (left_label);
+            box.append (link_image);
+
+            child = box;
+            add_css_class ("link");
+        }
+
+        public void launch (AppStream.Component component) {
+            if (uri_or_key.has_prefix ("http") || uri_or_key.has_prefix ("mailto")) {
+                var uri_launcher = new Gtk.UriLauncher (uri_or_key);
+                uri_launcher.launch.begin (
+                    ((Gtk.Application) GLib.Application.get_default ()).active_window,
+                    null
+                );
+                return;
+            }
+
+            var component_id = component.id;
+            if (component_id.has_suffix (".desktop")) {
+                // ".desktop" is always 8 bytes in UTF-8 so we can just chop 8 bytes off the end
+                component_id = component_id.substring (0, component_id.length - 8);
+            }
+
+            var stripe_dialog = new Widgets.StripeDialog (
+                1,
+                component.get_name (),
+                component_id,
+                uri_or_key
+            ) {
+                modal = true,
+                transient_for = ((Gtk.Application) Application.get_default ()).active_window
+            };
+
+            stripe_dialog.download_requested.connect (() => {
+                if (stripe_dialog.amount != 0) {
```

Comment:
This will cause a mem leak :(
(Using a local variable inside a lambda that's connected to a signal)

Reply:
Moved this bit to the stripe dialog itself which is more DRY anyways